### ISSUE TITLE
Plotting bug in pygmo.plotting.plot_non_dominated_fronts

### ISSUE THREAD

**Issue by horychen** [esa/pagmo2#301]
**1. My tentative fix** 
In the file "pagmo2/pygmo/plotting/__init__.py", at Line 77 and 78, 
**change** [points[idx][0] **to** [points[idx][comp[0]], 
and also **change** [points[idx][1] **to** [points[idx][comp[1]].

For your convenience, I paste line 70-78 from "pagmo2/pygmo/plotting/__init__.py" here and the link below:
```python
    for ndr, front in enumerate(fronts):
        # We plot the points
        for idx in front:
            ax.plot(points[idx][comp[0]], points[idx][
                comp[1]], marker=marker, color=cl[ndr])
        # We plot the fronts
        # Frist compute the points coordinates
        x = [points[idx][0] for idx in front] #<-----------------Line 77
        y = [points[idx][1] for idx in front] #<-----------------Line 78
```

> https://github.com/esa/pagmo2/blob/b8edfc2e86357a48e9051fb0597d43b276472fd0/pygmo/plotting/__init__.py

**2. How I found this bug** 
I was working on a multi-objective problem with 7 free variables and 3 objectives (denoted by OA, OB, OC). Somehow I managed to get an instance of pg.population as pop. Then I wanted to see the Pareto front surface projected to the 2D plane (e.g., OB vs. OA), so I ran the script as follows:
```pyrhon
        fits, vectors = pop.get_f(), pop.get_x()
        ax = pg.plot_non_dominated_fronts(fits, comp=[0,1], marker='o') # works as expected
        ax = pg.plot_non_dominated_fronts(fits, comp=[0,2], marker='o') # not expected results
        ax = pg.plot_non_dominated_fronts(fits, comp=[1,2], marker='o') # not expected results
```
About the outcome, please see the figure below. 
The subplot on the left corresponds to comp=[0, 1], which is perfect.
The subplot on the right corresponds to comp=[0,2], which is wrong.
![image](https://user-images.githubusercontent.com/44207479/59060124-085f0f00-8866-11e9-80e3-3012d162fdaf.png)
Sorry I don't know an easy way to give a working example, but I believe you get what I mean.

----
**3.** Another thing is about calling plt.show() inside this function (line 86). It makes the example clean, but maybe it would be better to leave this to the user, so that the user can actually add title and labels to the 
plot via ax.
```python
    plt.show()
    return ax
```

----
**4.** By the way, I have another unrelated question. After you make a change to the project pagmo2, how can I receive such change on my PC? I have tried to reinstall pygmo via conda as follows:
```
conda install pygmo --force-reinstall
```
but it is not working, of course. I am quite new to this "Github and conda package manager" world. I guess I will have to wait until the awesome guys behind pagmo2 release a new release of the software, right?

----
**5.** I will cite your software in my work. Just to make sure, is this bibtex citation correct?
@misc{francesco_biscani_2019_2529931,
  author       = {Francesco Biscani and
                  Dario Izzo and
                  Wenzel Jakob and
                  Marcus Märtens and
                  Alessio Mereta and
                  Cord Kaldemeyer and
                  Sergey Lyskov and
                  Sylvain Corlay and
                  Benjamin Pritchard and
                  Kishan Manani and
                  Johan Mabille and
                  Tomasz Miąsko and
                  Axel Huebl and
                  jakirkham and
                  hulucc and
                  polygon and
                  Zihao Fu and
                  The Gitter Badger and
                  Merlin Nimier-David and
                  Luka Čehovin Zajc and
                  Jonas Adler and
                  John Travers and
                  Jeongseok Lee and
                  Jakob Jordan and
                  Ivan Smirnov and
                  Huu Nguyen and
                  Felipe Lema and
                  Erik O'Leary and
                  Andrea Mambrini},
  title        = {esa/pagmo2: pagmo 2.10},
  month        = jan,
  year         = 2019,
  doi          = {10.5281/zenodo.2529931},
  url          = {https://doi.org/10.5281/zenodo.2529931}
}

**Comment by bluescarni** [comment]
@horychen thanks for the report, this looks like a real bug. Pinging @darioizzo as he originally wrote the code (he will be a bit busy for a few more days, but the fix should be in for the next version).

We fixed the data in the DOI, as it was containing some extraneous entries. The data at https://doi.org/10.5281/zenodo.2529931 should now be correct.

**Comment by bluescarni** [comment]
> By the way, I have another unrelated question. After you make a change to the project pagmo2, how can I receive such change on my PC?

If you are using pagmo via conda, you'll have to wait for the next official release. This is planned towards the end of the month.

**Comment by bluescarni** [comment]
@horychen this should be fixed by #301. The fix will be in the next pagmo version. Please re-open if you still experience the issue.
